integration_tests/str_to_list_cast.py

```diff
@@ -0,0 +1,14 @@
+def f():
+    x: list[str]
+    x = list("lpython")
+    s: str
+    s = "lpython"
+    i: i32
+    for i in range(len(s)):
+        assert x[i] == s[i]
+    x = list("")
+    assert len(x) == 0
+    x = list("L")
+    assert len(x) == 1 and x[0] == 'L'
+
+f()
```

Comment:
This is okay. Really surprising though. None of the variables (`s`, `x`) are marked as constant but still they work with only the compile time logic you have added.

Reply:
That is because if we have a string value at compile time then we can directly make a list constant out of it. So no cast, in that case, is required.